File: interviewBit/majorityElement.cpp
int Solution::majorityElement(const vector<int> &A) {
    int n = A.size();
    int ele = 0;
    int count = 0;
    for(int i=0;i<n;i++){
        if(count==0){
            ele = A[i];
        }
        if(ele == A[i]){
            count++;
        }else{
            count--;
        }
    }
    return ele;
}

File: basics/majorityElement.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        freq={}
        for i in range(len(nums)):
            if nums[i] in freq:
                freq[nums[i]]+=1
            else:
                freq[nums[i]]=1
        
        for key,value in freq.items():
            if value > len(nums)/2:
                return key
                
## optimize code
class Soltution:
    def majorityElement(self,nums):
        nums.sort()
        n = len(nums)
        return nums[n//2]

